Failure on non x86 arches (armv7/aarch64)

This builds on x86_64 for some reason but fails when building on Arm, either aarch64 or ARMv7.

<pre>
BUILDSTDERR: main.cpp: In function 'bool upgrade_loader(STRUCT_RKDEVICE_DESC&, char*)':
BUILDSTDERR: main.cpp:1915:13: error: comparison is always false due to limited range of data type [-Werror=type-limits]
BUILDSTDERR:  1915 |   if (index == -1) {
BUILDSTDERR:       |       ~~~~~~^~~~~
BUILDSTDERR: main.cpp:1939:13: error: comparison is always false due to limited range of data type [-Werror=type-limits]
BUILDSTDERR:  1939 |   if (index == -1) {
BUILDSTDERR:       |       ~~~~~~^~~~~
BUILDSTDERR: cc1plus: all warnings being treated as errors
</pre>

--- a/main.cpp
+++ b/main.cpp
@@ -1882,7 +1882,7 @@ bool upgrade_loader(STRUCT_RKDEVICE_DESC &dev, char *szLoader)
 	CRKComm *pComm = NULL;
 	bool bRet, bSuccess = false;
 	int iRet;
-	char index;
+	signed char index;
 	USHORT usFlashDataSec, usFlashBootSec;
 	DWORD dwLoaderSize, dwLoaderDataSize, dwDelay, dwSectorNum;
 	char loaderCodeName[] = "FlashBoot";

